tsl/src/nodes/skip_scan/planner.c

```diff
@@ -372,25 +382,108 @@ skip_scan_path_create(PlannerInfo *root, IndexPath *index_path, double ndistinct
 	 * free so reusing the IndexPath here is safe. */
 	skip_scan_path->index_path = index_path;
 
-	/* find the ordering operator we'll use to skip around each key column
-	 * Since the DISTINCT columns are required to be prefix of ORDER BY
-	 * and we only support single column DISTINCT the first pathkey
-	 * has to be the distinct column
-	 */
-	PathKey *first_pathkey = linitial(index_path->path.pathkeys);
-	TargetEntry *tle = get_tle_for_pathkey(index_path->indexinfo->indextlist, first_pathkey, true);
+	Var *var = get_distinct_var(root, index_path, skip_scan_path);
 
-	/* SkipScan on expressions not supported */
-	if (!tle || !IsA(tle->expr, Var))
+	if (!var)
 		return NULL;
 
 	/* build skip qual this may fail if we cannot look up the operator */
-	if (!build_skip_qual(root, skip_scan_path, index_path, castNode(Var, tle->expr)))
+	if (!build_skip_qual(root, skip_scan_path, index_path, var))
 		return NULL;
 
 	return skip_scan_path;
 }
 
+/* Extract the Var to use for the SkipScan and do the TupleConversion
+ * if required.
+ */
+static Var *
+get_distinct_var(PlannerInfo *root, IndexPath *index_path, SkipScanPath *skip_scan_path)
```

Comment:
You also update the skip scan path. You should make that clear through, e.g., naming. Otherwise, it is assumed that this is a pure getter, which it is not.